pkg/cmd/roachtest/tests/copyfrom.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/stretchr/testify/require"
+)
+
+const tpchLineitemFmt = `https://storage.googleapis.com/cockroach-fixtures/tpch-csv/sf-%d/lineitem.tbl.1`
+
+// There's an extra dummy field because the file above ends lines with delimiter and standard CSV behavior is to
+// interpret that as a column.
+var lineitemSchema string = `
+CREATE TABLE  lineitem (
+	l_orderkey      INT8 NOT NULL,
+	l_partkey       INT8 NOT NULL,
+	l_suppkey       INT8 NOT NULL,
+	l_linenumber    INT8 NOT NULL,
+	l_quantity      DECIMAL(15,2) NOT NULL,
+	l_extendedprice DECIMAL(15,2) NOT NULL,
+	l_discount      DECIMAL(15,2) NOT NULL,
+	l_tax           DECIMAL(15,2) NOT NULL,
+	l_returnflag    CHAR(1) NOT NULL,
+	l_linestatus    CHAR(1) NOT NULL,
+	l_shipdate      DATE NOT NULL,
+	l_commitdate    DATE NOT NULL,
+	l_receiptdate   DATE NOT NULL,
+	l_shipinstruct  CHAR(25) NOT NULL,
+	l_shipmode      CHAR(10) NOT NULL,
+	l_comment       VARCHAR(44) NOT NULL,
+	l_dummy         CHAR(1),
+	PRIMARY KEY     (l_orderkey, l_linenumber));
+CREATE INDEX  l_ok ON lineitem    (l_orderkey);
+CREATE INDEX  l_pk ON lineitem    (l_partkey);
+CREATE INDEX  l_sk ON lineitem    (l_suppkey);
+CREATE INDEX  l_sd ON lineitem    (l_shipdate);
+CREATE INDEX  l_cd ON lineitem    (l_commitdate);
+CREATE INDEX  l_rd ON lineitem    (l_receiptdate);
+CREATE INDEX  l_pk_sk ON lineitem (l_partkey, l_suppkey);
+CREATE INDEX  l_sk_pk ON lineitem (l_suppkey, l_partkey);
+`
+
+func initTest(ctx context.Context, t test.Test, c cluster.Cluster, sf int) {
+	if runtime.GOOS == "linux" {
+		if err := repeatRunE(
+			ctx, t, c, c.All(), "update apt-get", `sudo apt-get -qq update`,
+		); err != nil {
+			t.Fatal(err)
+		}
+		if err := repeatRunE(
+			ctx,
+			t,
+			c,
+			c.All(),
+			"install dependencies",
+			`sudo apt-get install -qq postgresql`,
```

Comment:
do you think it's worth using a stable postgresql version?

Reply:
I can't think of a reason why, ubuntu LTS should be stable enough no?   Open to suggestions...

Reply:
In the past we've found that the package archive increases if you do sudo apt-get install and then you get result changes that don't correlate. If you wanted to be a bit more particular I'd install from source, but I am not going to hold this pr down for it.